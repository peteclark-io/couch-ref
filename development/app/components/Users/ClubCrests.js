'use strict';

import React from 'react';
import {connect} from 'react-redux';

import {ThreeBounce} from 'better-react-spinkit';

import {selectClub} from '../../ducks/user';
import styles from './styles.css';
import {saveClub} from '../../core/cookies';

const ClubCrests = React.createClass({

  propTypes: {
    clubs: React.PropTypes.array,
    selectClub: React.PropTypes.func
  },

  contextTypes: {
     router: React.PropTypes.object
  },

  render: function() {
    if (!this.props.clubs){
      return (
         <div className={styles.loading}>
           <ThreeBounce color="white" />
         </div>
      );
    }

    return (
      <div>
         <h3 className={styles.question}>Who Do You Support in the Premier League?</h3>
         <section className={styles['clubs-container']}>
           <ul className={styles.clubs}>
           {this.props.clubs.map((c) => {
              return (
                 <li key={c.name}>
                 <img src={c.crestUrl} alt={c.name} onClick={() => {
                    this.props.selectClub(c);
                    this.context.router.push('/users/dob');
                 }}></img>
                 </li>
              );
           })}
           </ul>
           {/*<button>Other</button>*/}
           <h4 className={styles.disclaimer}>We currently only collect data on Premier League fans, sorry if you support someone else!</h4>
         </section>
      </div>
    );
  }
});

const getClubs = (state = {clubs: []}, id) => {
  return state.clubs.length > 0 ? state.clubs : null;
};

const mapStateToProps = (state) => {
   return {
     clubs: getClubs(state)
   };
};

const mapDispatchToProps = (dispatch) => {
  return {
    selectClub: (club) => {
      dispatch(selectClub(club));
      saveClub(club);
    }
  }
};

const LiveClubCrests = connect(
  mapStateToProps,
  mapDispatchToProps
)(ClubCrests);

export default LiveClubCrests;
