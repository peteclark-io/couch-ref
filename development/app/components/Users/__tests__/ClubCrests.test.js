'use strict';

import React from 'react';
import {ClubCrests} from '../ClubCrests';
import renderer from 'react-test-renderer';
import {mount, shallow} from 'enzyme';

it('Shows loading screen when no data.', () => {
   const rendered = shallow(
      <ClubCrests />
   );
   expect(rendered.find('.loading')).toHaveLength(1);
});

it('Renders ok with clubs.', () => {
   const clubs = [{url: 'url', name: 'Arsenal'}];
   const user = {name: 'pedro'};

   var routed = false;
   const router = {push: (path) => {
      routed = true;
      expect(path).toEqual('/users/birthday');
   }}

   var clicked = false;
   const rendered = mount(
      <ClubCrests location={{query: {edited: false}}} clubs={clubs} user={user} selectClub={(user, club) => {
         clicked = true;
         expect(user.name).toEqual('pedro');
         expect(club.name).toEqual('Arsenal');
         expect(club.url).toEqual('url');
      }} />,
      {context: {router: router}}
   );

   expect(rendered.html()).toMatchSnapshot();
   rendered.find('img[alt="Arsenal"]').simulate('click');

   expect(routed).toBeTruthy();
   expect(clicked).toBeTruthy();
});

it('Should return to the edit page if edited.', () => {
   const clubs = [{url: 'url', name: 'Arsenal'}];
   const user = {name: 'pedro'};

   var routed = false;
   const router = {push: (path) => {
      routed = true;
      expect(path).toEqual('/users/edit');
   }}

   var clicked = false;
   const rendered = mount(
      <ClubCrests location={{query: {edited: true}}} clubs={clubs} user={user} selectClub={(user, club) => {
         clicked = true;
         expect(user.name).toEqual('pedro');
         expect(club.name).toEqual('Arsenal');
         expect(club.url).toEqual('url');
      }} />,
      {context: {router: router}}
   );

   expect(rendered.html()).toMatchSnapshot();
   rendered.find('img[alt="Arsenal"]').simulate('click');

   expect(routed).toBeTruthy();
   expect(clicked).toBeTruthy();
});
