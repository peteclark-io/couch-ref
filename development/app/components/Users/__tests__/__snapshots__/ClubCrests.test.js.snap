exports[`test Renders ok with clubs. 1`] = `"<div><h3 class=\"question\">Who Do You Support in the Premier League?</h3><section class=\"clubs-container\"><ul class=\"clubs\"><li><img alt=\"Arsenal\"></li></ul><h3>Other Clubs</h3><ul class=\"clubs\"><li><a class=\"other-users\">EFL Championship</a></li><li><a class=\"other-users\">League One</a></li><li><a class=\"other-users\">League Two</a></li><li><a class=\"other-users\">National League</a></li><li><a class=\"other-users\">Other</a></li></ul><h4 class=\"disclaimer\">We currently only collect data on Premier League clubs, sorry if you support someone else! Please select which league the club you support belongs to above.</h4><div class=\"spacer\"></div></section></div>"`;
